### ISSUE TITLE
In full-screen page tearsheet view, I want the cancel dialog window to appear only if any form changes are made.

### ISSUE THREAD

**Issue by angel-neethuej** [carbon-design-system/ibm-products#6590]
I am using a [full-page carbon tearsheet](https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-patterns-create-flows-createfullpage--create-full-page&globals=viewport:basic) view, and when I click the Cancel button, the confirmation dialog should only appear if the form has changed... if there hasn't, I want to be redirected to the previous page without seeing the confirmation dialog!
In short, the dev should have some control over whether the dialog confirmation should be displayed or not.
 ![Image](https://github.com/user-attachments/assets/ff980ff9-055a-433a-8efa-08195e2e5728)

**Comment by davidmenendez** [comment]
as a followup to the discussion in slack:

> it's here [https://github.com/carbon-design-system/ibm-products/blob/9595f65c59af0e25330b2c66[…]s/ibm-products/src/components/CreateFullPage/CreateFullPage.tsx](https://github.com/carbon-design-system/ibm-products/blob/9595f65c59af0e25330b2c6674ebc8ba6e68c5fb/packages/ibm-products/src/components/CreateFullPage/CreateFullPage.tsx#L403-L411)
[8:41](https://ibm-studios.slack.com/archives/C013ZTX0N6B/p1734446467331399?thread_ts=1734441927.482939&cid=C013ZTX0N6B)
and here [https://github.com/carbon-design-system/ibm-products/blob/9595f65c59af0e25330b2c66[…]s/ibm-products/src/components/CreateFullPage/CreateFullPage.tsx](https://github.com/carbon-design-system/ibm-products/blob/9595f65c59af0e25330b2c6674ebc8ba6e68c5fb/packages/ibm-products/src/components/CreateFullPage/CreateFullPage.tsx#L417-L424)
[8:41](https://ibm-studios.slack.com/archives/C013ZTX0N6B/p1734446493162709?thread_ts=1734441927.482939&cid=C013ZTX0N6B)
basically setModalIsOpen(!modalIsOpen); is always called in create full page when cancel / close is called

**Comment by amal-k-joy** [comment]
@angel-neethuej 
We have created [createFullPage](https://ibm-products.carbondesignsystem.com/?path=/docs/patterns-create-flows-create-full-page--overview) as a true pattern instead as a pattern component. This is available in the storybook. The idea behind the  true pattern is that we will share you the complete experience in a stackblitz app and you can consume it from there and you can customise it to achieve this functionality.

Please feel free to re-open if needed.
